docs/create/guides/nft-project/dapp.mdx

```diff
@@ -0,0 +1,213 @@
+---
+sidebar_position: 4
+title: Building the NFT dApp
+---
+
+If it's is your first time building an Archway dApp frontend, head over to the [Creating Your First dApp](../my-first-dapp/dapp.mdx) guide to learn how to setup your basic project scaffolding. Once you're serving a web page that connects to [Keplr](https://wallet.keplr.app/), you'll be ready to follow the next steps.
+
+## Loading NFT the collection
+
+As we saw in the previous guide, to query a contract we need to know its address on chain. We also need an instance of `SigningCosmWasmClient` from `@cosmjs/cosmwasm-stargate` that's been connected to an Archway network such as [Constantine testnet](https://rpc.constantine-1.archway.tech/).
+
+```js
+// For more on setting up this handler see the "Creating Your First dApp" guide (linked to above):
+const queryHandler = CosmWasmClient.queryClient.wasm.queryContractSmart;
+// Replace this empty string with your deployed contract address
+const contract = "";
+```
+
+Using the above two variables we can write JavaScript to query the NFT collection for its token ids like this:
+
+```js
+async function getNfts () {
+  let entrypoint = {
+    all_tokens: {}
+  };
+  let query = await queryHandler(contract, entrypoint);
+  console.log('All NFTs of contract ' + contract + ':', query);
+  return query;
+}
+```
+
+The `getNfts` function returns a `tokens` object containing an array of all tokens minted by the contract. 
+
+Here's an example of calling it:
+
+```js
+let nfts;
+async function doGetNfts() {
+  nfts = await getNfts();
+  console.log('All NFT ids', nfts);
+}
+doGetNfts();
+```
+
+## Minting from the dApp
+
+To mint from our dApp, we need to assemble the metadata fields for the NFT. We can achieve this by making a web UI with form fields where the NFT creator can add custom traits to their NFT.
+
+<!-- [screenshot goes here] -->
+
+We also need the `calculateFee` and `GasPrice` functions from the [@cosmjs/stargate](https://www.npmjs.com/package/@cosmjs/stargate) package.
+
+```js
+const gasPrice = GasPrice.fromString('0.002uconst');
+const txFee = calculateFee(300000, this.gas.price);
+```
+
+We can simplify our remaining task by writing a function which takes the required NFT metadata as parameters. With that in mind, here's what a minting function looks like in JavaScript:
+
+```js
+async function mintNft (owner = null, tokenId = null, imageUrl = null, name = null, description = null, attributes = []) {
+  // Ensure we collected the required arguments
+  if (!owner || !tokenId || !imageUrl || !name || !description) {
+    console.warn('Error parsing mint arguments', {owner: owner, tokenId: tokenId, imageUrl: imageUrl, name: name, description: description});
+    return;
+  }
```

Comment:
To improve the readability and avoid horizontal scrolling, I'd suggest removing the `null` from the arguments since they are, by default, `undefined`. You can also enhance the map object by using the shorthand initializer (it's safe to use in all browsers).

```suggestion
async function mintNft(owner, tokenId, imageUrl, name, description, attributes = []) {
  // Ensure we collected the required arguments
  if (!owner || !tokenId || !imageUrl || !name || !description) {
    console.warn('Error parsing mint arguments', { owner, tokenId, imageUrl, name, description });
    return;
  }
```